File: rust/src/models/big_query_config_patch.rs
// this file is @generated
#[allow(unused_imports)]
use js_option::JsOption;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct BigQueryConfigPatch {
    #[serde(rename = "projectId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub project_id: Option<String>,

    #[serde(rename = "datasetId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub dataset_id: Option<String>,

    #[serde(rename = "tableId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub table_id: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub credentials: Option<String>,
}

impl BigQueryConfigPatch {
    pub fn new() -> Self {
        Self {
            project_id: None,
            dataset_id: None,
            table_id: None,
            credentials: None,
        }
    }
}

impl Default for BigQueryConfigPatch {
    fn default() -> Self {
        Self::new()
    }
}

File: javascript/src/models/bigQueryConfigPatch.ts
// this file is @generated

export interface BigQueryConfigPatch {
  projectId?: string;
  datasetId?: string;
  tableId?: string;
  credentials?: string;
}

export const BigQueryConfigPatchSerializer = {
  _fromJsonObject(object: any): BigQueryConfigPatch {
    return {
      projectId: object["projectId"],
      datasetId: object["datasetId"],
      tableId: object["tableId"],
      credentials: object["credentials"],
    };
  },

  _toJsonObject(self: BigQueryConfigPatch): any {
    return {
      projectId: self.projectId,
      datasetId: self.datasetId,
      tableId: self.tableId,
      credentials: self.credentials,
    };
  },
};
